libyul/AsmAnalysis.cpp

```diff
@@ -772,10 +772,27 @@ bool AsmAnalyzer::validateInstructions(evmasm::Instruction _instr, SourceLocatio
 			"PC instruction is a low-level EVM feature. "
 			"Because of that PC is disallowed in strict assembly."
 		);
+	else if (!m_eofVersion.has_value() && (
+		_instr == evmasm::Instruction::EXTCALL ||
+		_instr == evmasm::Instruction::EXTDELEGATECALL ||
+		_instr == evmasm::Instruction::EXTSTATICCALL
+	))
+	{
+		m_errorReporter.typeError(
+			4328_error,
+			_location,
+			fmt::format(
+				"The \"{instruction}\" instruction is {kind} VMs.",
+				fmt::arg("instruction", boost::to_lower_copy(instructionInfo(_instr, m_evmVersion).name)),
+				fmt::arg("kind", "not available in legacy bytecode")
+			)
+		);
+	}
 	else if (m_eofVersion.has_value() && (
 		_instr == evmasm::Instruction::CALL ||
 		_instr == evmasm::Instruction::CALLCODE ||
 		_instr == evmasm::Instruction::DELEGATECALL ||
+		_instr == evmasm::Instruction::STATICCALL ||
```

Comment:
Will move to separate. We have https://github.com/ethereum/solidity/pull/15559/files#diff-30c103e3872c974dedcee26006d5ffad9eca618467cc82a4f7f1aed465c16fae for legacy calls.

Reply:
> We have https://github.com/ethereum/solidity/pull/15559/files#diff-30c103e3872c974dedcee26006d5ffad9eca618467cc82a4f7f1aed465c16fae for legacy calls.


The link does not point me at any specific file... I guess you meant that we already have a test covering all legacy-only opcodes? Which one?

Reply:
Ok, that covers this specific case, but I still think we need one that just covers all the opcodes, because we keep running into bugs like this.

Along the same lines as what I just suggested in https://github.com/ethereum/solidity/pull/15559#discussion_r1873671693, but for legacy-only opcodes.

Reply:
Just to be clear, does not have to be done in this PR. I guess we can continue this discussion in #15620.